fuse/data/tokenizers/modular_tokenizer/modular_tokenizer.py

```diff
@@ -1214,23 +1238,33 @@ def encode(
         ), f"Error: expecting leading modular tokenizer hints followed by a sequence to tokenize, got {sequence}"
         # arrange as a list of TypedInput - each one will include the type and the following sequence
         encode_list_format = []
-        for tokenizer_type, subseq in zip(
+        for tokenizer_hints, subseq in zip(
             hints_and_subseq[::2], hints_and_subseq[1::2]
         ):
-            max_len_str = "@MAX-LEN="
-            curr_max_len_idx = tokenizer_type.find(max_len_str)
-            if curr_max_len_idx > 0:
-                curr_max_len = tokenizer_type[curr_max_len_idx + len(max_len_str) :]
-                try:
-                    curr_max_len = int(curr_max_len)
-                except:
-                    raise Exception(
-                        f"Had a problem casting curr_max_len={curr_max_len} to int! it was found inside modular tokenizer meta TOKENIZER-TYPE={tokenizer_type}"
-                    )
-                tokenizer_type = tokenizer_type[:curr_max_len_idx]
-            else:
-                curr_max_len = None
-            encode_list_format.append(TypedInput(tokenizer_type, subseq, curr_max_len))
+            tokenizer_hints_parts = tokenizer_hints.split("@")
+            tokenizer_type = tokenizer_hints_parts[0]
+            curr_max_len = None
+            curr_truncate_mode = None
+            for part in tokenizer_hints_parts[1:]:
+                if part.startswith("MAX-LEN="):
+                    try:
+                        curr_max_len = int(part[len("MAX-LEN=") :])
+                    except:
+                        raise Exception(
+                            f"Had a problem casting curr_max_len={part} to int! it was found inside modular tokenizer meta TOKENIZER-TYPE={tokenizer_hints}"
+                        )
+                elif part.startswith("TRUNC-MODE="):
```

Comment:
can this be added in some documentation somewhere ?

Reply:
Good point - will add to the README file.

Reply:
Actually, this all part, about meta tokens and TyoedInput is missing and should be added to README.
I'm opening a task for it.
Maybe we can use the text from the paper to improve the README of the modular tokenizer.